lib/blockchain.go

```diff
@@ -2639,9 +2639,170 @@ func (bc *Blockchain) processBlockPoW(desoBlock *MsgDeSoBlock, verifySignatures
 // 4. Connect the block to the blockchain's tip
 // 5. If applicable, flush the incoming block's grandparent to the DB
 // 6. Notify the block proposer, pacemaker, and voting logic that the incoming block has been accepted
-func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, verifySignatures bool) (_isMainChain bool, _isOrphan bool, err error) {
+func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, verifySignatures bool) (_success bool, _isOrphan bool, _blockHashToRequest *BlockHash, _err error) {
 	// TODO: Implement me
-	return false, false, fmt.Errorf("ProcessBlockPoS: Not implemented yet")
+	// 1. Surface Level validation fo the block
```

Comment:
Suggestion. Maybe we could simplify this blockchain.go a little by consolidating the sanity checks into one `validateDeSoBlockPos()`, or something alike. For pos mempool I made a special file called `validation_deso_txn` where I confined all the validation logic for MsgDeSoTxn. We could do the same for MsgDeSoBlock with a file like `validation_deso_block`. Usually these sanity checks only require a couple of inputs like desoparams or globalparams, so it feels like they these can be made static functions. In the case of mempool, I think the code ended up much simpler, and txn validation was easily portable to other places like UtxoView. What do you think?

Reply:
I've moved it to a new file and packaged it all together into one calling function. Having a bunch of little functions allows for easier unit testing and for breaking the work into smaller chunks.

Unfortunately we still need access to the blockchain struct in order to perform these validations as we are comparing attributes of the block to the current state of the blockchain, so reuse won't be that valuable unfortunately. but certainly feels nice to have them in a separate file.